### ISSUE TITLE
Add the requestBody for the post endpoints

### ISSUE THREAD

**Issue by caduvieira** [CachetHQ/Docs#11]
The post endpoints in the API doc doesn't have a requestBody example or specification. Would be nice to have it.

If added to the openapi.yaml should be enough
https://learn.openapis.org/specification/parameters.html

**Comment by jbrooksuk** [comment]
Great spot! I'm not at my laptop at the moment, but I'll get this fixed when I'm back.

**Comment by jbrooksuk** [comment]
I had a quick look at this earlier and it's caused by us switching to `spatie/laravel-data`. Scribe is no longer able to read a `FormRequest` so is unable to derive the request body.

We'll need to write a custom Scribe strategy to pull the data from Laravel Data, pretty gnarly.

**Comment by jbrooksuk** [comment]
Actually, pinging @joelbutcher (refactored Cachet into `laravel-data` and @olivernybroe (experience writing Scribe strategies).

**Comment by joelbutcher** [comment]
@jbrooksuk I'm working on a Scribe strategy specifically for Laravel Data for work. I'm hoping to publish this soon. If you assign this to me, I'll make sure to create a PR once this is done. 👍

**Comment by jbrooksuk** [comment]
Hah, no way @joelbutcher! I'll make the PR in Cachet itself that installs, configures and generates the Scribe docs.

**Comment by joelbutcher** [comment]
@jbrooksuk https://github.com/joelbutcher/scribe-laravel-data-plugin

**Comment by jbrooksuk** [comment]
@joelbutcher I've tested this out (https://github.com/cachethq/cachet/pull/4449) but it's not generating everything. A good example is the `POST components` endpoint. Only the name field is generated.

**Comment by joelbutcher** [comment]
@jbrooksuk That's odd, by the looks of it, Laravel Data itself isn't returning any rules for the properties with default values. Might have to dig a bit deeper into laravel-data as to why this is occurring. An alternative is to scrap using PHP attributes and define a static `rules` method on each request (similar to Laravel Form requests):

```php
public static function rules(ValidationContext $context): array
{
    return [
        //...
    ];
}
```

**Comment by joelbutcher** [comment]
@jbrooksuk I'll send in a PR today that uses the `rules` method to explicitly define the validation rules, rather than having Laravel Data infer from types and attributes

**Comment by jbrooksuk** [comment]
Amazing, thanks @joelbutcher!

**Comment by joelbutcher** [comment]
@jbrooksuk at one point we had `float` as a validation rule for creating and updating a metric, but I'm now running into this error:

```bash
Method Illuminate\Validation\Validator::validateFloat does not exist.
```

Is there an alternative rule we can use? I was thinking [`decimal`](https://laravel.com/docs/10.x/validation#rule-decimal)?

**Comment by jbrooksuk** [comment]
Yeah, `decimal` should be fine.
